### PR TITLE
[M3] Migrate CI to Pipenv (#59)

### PR DESCRIPTION
**Pull request by ayoubmansar**
This pull request solves issue #59 — see the issue for more information.

This PR modifies the `main.yml` file which dictates to Github Actions what to run. I've also made some minor style updates so that each of the commands are properly grouped with a corresponding name and removed the redundant `cd src && ...` pattern in each of the commands. I've removed the old `requirements.txt` file in favor of the newly generated `Pipfile.lock` file 💯 

_Pipenv now passing CI_
Screen Shot 2023-03-28 at 4 32 03 AM

_CI step names are now insightful_
Screen Shot 2023-03-28 at 4 33 03 AM

### CODE REVIEW

File: src/SearchEngine/WebpageSearcher.py [hunk 1]
```diff
@@ -96,16 +128,23 @@ def preprocess_text(text):
...
     # Scrape the webpage
+    print("here1")
     page = requests.get(url)
+    print("here2")
     soup = BeautifulSoup(page.content, 'html.parser')
+    print("here3")
 
     # Extract the visible text from the webpage
     texts = soup.findAll(text=True)
+    print("here4")
```
[the remark this answers is not in the document]
Comment (ayoubmansar) [reply src/SearchEngine/WebpageSearcher.py:139]: This commit is based on top of Ashwin's branch. Those debugging print statements were added by him (da2acdc).

I've already let Ashwin know in Slack to please remove all of these print statements, once he does that I can rebase on top of that and it should automatically update the PR.

**Review by teddyliang — COMMENTED** [review]
Looks good, could you update the README.md with the updated instructions on how to run the app?

**Comment by ayoubmansar** [comment]
Updated `README.md` as well

**Review by teddyliang — APPROVED** [review]
Looks good

**Pull request merged**